### PR TITLE
Csv upgrade

### PR DESCRIPTION
**Pull request by WChrisK**
Now does: `reviewerGroup, revieweeGroup [, revieweeGroup...]` and fixed a bug.

### CODE REVIEW

File: app/controllers/peer_reviews_controller.rb [hunk 1]
```diff
@@ -107,18 +115,21 @@ def download_reviewer_reviewee_mapping
...
-        raise CSVInvalidLineError if row.empty?
-        result = Result.find(row.first)
-        reviewer = Grouping.joins(:group).find_by(
-                                groups: { group_name: row.second },
-                                assignment_id: assignment_id)
-        PeerReview.create!(result: result, reviewer: reviewer)
+      result = MarkusCSV.parse(pr_mapping.read, encoding: encoding) do |row|
+        raise CSVInvalidLineError if row.size < 2
+        reviewer = Grouping.joins(:group).find_by(groups: { group_name: row.first })
+        row.shift()  # Drop the reviewer, the rest are reviewees and makes iteration easier.
```
Comment (david-yz-liu) [inline app/controllers/peer_reviews_controller.rb]: You don't need the `()`

File: app/controllers/peer_reviews_controller.rb [hunk 2]
```diff
@@ -107,18 +115,21 @@ def download_reviewer_reviewee_mapping
...
-                               encoding: params[:encoding]) do |row|
-        raise CSVInvalidLineError if row.empty?
-        result = Result.find(row.first)
-        reviewer = Grouping.joins(:group).find_by(
-                                groups: { group_name: row.second },
-                                assignment_id: assignment_id)
-        PeerReview.create!(result: result, reviewer: reviewer)
+      result = MarkusCSV.parse(pr_mapping.read, encoding: encoding) do |row|
+        raise CSVInvalidLineError if row.size < 2
+        reviewer = Grouping.joins(:group).find_by(groups: { group_name: row.first })
```
Comment (david-yz-liu) [inline app/controllers/peer_reviews_controller.rb]: This logic isn't quite right if the same group has groupings for multiple assignments.

First get the `Group`, then use the `grouping_for_assignment` method to get the right grouping. Applies below as well.

File: lib/tasks/peer_review.rake [hunk 3]
```diff
@@ -8,8 +8,17 @@ namespace :db do
     a1 = Assignment.find_by(short_identifier: 'A1')
     a1.update(has_peer_review: true)  # Creates 'a1pr' via callback.
 
+    # The group names must not be closed, doing a1pr.clone_groupings_from(a1.id)
```
Comment (david-yz-liu) [inline lib/tasks/peer_review.rake]: I think you mean "cloned," but I also think that this is not a problem if you resolve my above comment.

**Comment by WChrisK** [comment]
After doing the change for using group names, it now works with the old clone method.

**Pull request merged**